app/models/base_linehaul.rb

```diff
@@ -0,0 +1,7 @@
+class BaseLinehaul < ApplicationRecord
+  def self.rate(date, distance, weight)
+    select(:rate).find_by('effective @> ?::date AND dist_mi @> ? AND weight_lbs @> ?', date, distance, weight).rate
+  end
+
+  validates :effective, :dist_mi, :weight_lbs, :rate, presence: true
```

Comment:
Order in a Rails model doesn't _entirely_ matter, but for consistency's sake, I've tried to put things in the following order:

1. Mixins
1. Miscellaneous stuff like constants, `friendly_id :foo`, etc.
1. Associations
1. Validations
1. Scopes
1. Callbacks
1. Instance methods
1. Class methods
1. Private methods

So in this case, you'd move the validation above the class method.